src/scipp/operations.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+from typing import Callable, Optional
+from inspect import signature
+from .core import Variable
+from ._scipp.core import experimental_transform
+
+
+def _as_numba_cfunc(function, unit_func=None):
+    import numba
+    dtype = 'double'
+    narg = len(signature(function).parameters)
+    cfunc = numba.cfunc(dtype + '(' + ','.join([dtype] * narg) + ')')(function)
+    cfunc.unit_func = function if unit_func is None else unit_func
+    return cfunc
+
+
+def elementwise_transform(func: Callable,
```

Comment:
`elemwise_transform`? `elemwise_func`?

I wanted to avoid the `data` part, since it may refer to the "data" of a data array, and we later want to do something like this:

```
func = sc.elemwise_func(myfunc)
da = sc.transform_data(da1, da2, da3, func)
```